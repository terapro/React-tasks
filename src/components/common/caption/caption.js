import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const Caption =(props) => {
  let itemClass = classNames({
    'txt': true,
    'txt-white': props.txtWhite
  });

  return (
    <p className={itemClass}>{props.content}</p>
  );
};
Caption.propTypes ={
  txtWhite: PropTypes.bool,
  content: PropTypes.string
};

export {Caption};