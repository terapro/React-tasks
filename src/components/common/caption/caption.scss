.txt {
  font-size: $textSize;
  line-height: 25px;
  padding: $elementPadding 0;
  text-align: justify;
}

.txt-white {
  color: $color_white;

}