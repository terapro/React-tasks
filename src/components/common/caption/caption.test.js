import React from 'react';
import {Caption} from 'src/components/common/caption/caption.js';
import {shallow} from 'enzyme';

const mockText = 'Text for test';

describe('<Caption />', () => {
  it('should render the caption', () => {
    const wrapper = shallow(<Caption />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render the caption content', () => {
    const wrapper = shallow(<Caption content={mockText}/>);
    expect(wrapper.text()).toBe(mockText);
  });
});