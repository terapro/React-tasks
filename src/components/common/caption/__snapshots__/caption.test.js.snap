// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Caption /> should render the caption 1`] = `
<p
  className="txt"
>
  Caption
</p>
`;
